hostinger/dns_record.go

```diff
@@ -0,0 +1,163 @@
+package hostinger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceHostingerDNSRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceHostingerDNSRecordCreate,
+		Read:   resourceHostingerDNSRecordRead,
+		Delete: resourceHostingerDNSRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  14400,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceHostingerDNSRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*HostingerClient)
+
+	zone := d.Get("zone").(string)
+	name := d.Get("name").(string)
+	recordType := d.Get("type").(string)
+	value := d.Get("value").(string)
+	ttl := d.Get("ttl").(int)
+
+	url := fmt.Sprintf("https://developers.hostinger.com/api/dns/v1/zones/%s/records", zone)
+
+	payload := map[string]interface{}{
+		"name":  name,
+		"type":  recordType,
+		"value": value,
+		"ttl":   ttl,
+	}
+	body, _ := json.Marshal(payload)
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Authorization", "Bearer "+client.Token)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	respBody, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode != 201 {
+		return fmt.Errorf("failed to create DNS record: %s", respBody)
+	}
+
+	var result map[string]interface{}
+	json.Unmarshal(respBody, &result)
+
+	id := fmt.Sprintf("%v", result["id"])
+	d.SetId(id)
+
+	return resourceHostingerDNSRecordRead(d, meta)
+}
+
+func resourceHostingerDNSRecordRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*HostingerClient)
+
+	zone := d.Get("zone").(string)
+	id := d.Id()
+
+	url := fmt.Sprintf("https://developers.hostinger.com/api/dns/v1/zones/%s/records", zone)
```

Comment:
Please reuse `addStandardHeaders` method for dynamic URL, dynamic User-agent and Barrier Token. Example:
```
func (c *HostingerClient) GetDefaultPaymentMethod() (int, error) {
	url := c.BaseURL + "/api/billing/v1/payment-methods"

	req, _ := http.NewRequest("GET", url, nil)
	c.addStandardHeaders(req)

	resp, err := c.HTTPClient.Do(req)
```